src/codegen/forward.jl

```diff
@@ -53,11 +53,12 @@ function fwd_transform!(ci, mi, nargs, N)
             return ZeroBundle{N}(mi.sparam_vals[stmt.args[1]::Int])
         elseif isexpr(stmt, :foreigncall)
             return Expr(:call, error, "Attempted to AD a foreigncall. Missing rule?")
-        elseif isexpr(stmt, :meta) || isexpr(stmt, :inbounds)
+        elseif isexpr(stmt, :meta) || isexpr(stmt, :inbounds)  || isexpr(stmt, :loopinfo)
             # Can't trust that meta annotations are still valid in the AD'd
             # version.
             return nothing
         else
+            #TODO put guard here. We really don't want invalid IR
```

Comment:
This is not in general invalid IR

Reply:
The proposal here is rather that we assert that this is only getting various kinds of literals to wrap in a `ZeroBundle` (be they functions or numbers etc)
and if not give a good error message, rather than generating invalid IR.